File: app/src/main/java/io/github/amadeu01/clone2048/view/MainActivity.java
package io.github.amadeu01.clone2048.view;

import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.KeyEvent;

import io.github.amadeu01.clone2048.R;
import io.github.amadeu01.clone2048.databinding.GameBinding;
import io.github.amadeu01.clone2048.model.Move;
import io.github.amadeu01.clone2048.viewmodel.GameViewModel;

public class MainActivity extends AppCompatActivity {

    GameViewModel viewModel = new GameViewModel();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        GameBinding binding = DataBindingUtil.setContentView(this, R.layout.game);
        binding.setViewModel(viewModel);
        viewModel.onCreate();
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_MENU) {
            //Do nothing
            return true;
        } else if (keyCode == KeyEvent.KEYCODE_DPAD_DOWN) {
            viewModel.onMove(Move.DOWN);
            return true;
        } else if (keyCode == KeyEvent.KEYCODE_DPAD_UP) {
            viewModel.onMove(Move.UP);
            return true;
        } else if (keyCode == KeyEvent.KEYCODE_DPAD_LEFT) {
            viewModel.onMove(Move.LEFT);
            return true;
        } else if (keyCode == KeyEvent.KEYCODE_DPAD_RIGHT) {
            viewModel.onMove(Move.RIGHT);
            return true;
        }

        return super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/me/amadeucavalcante/a2048/view/MainActivity.kt
package me.amadeucavalcante.a2048.view

import android.databinding.DataBindingUtil
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import me.amadeucavalcante.a2048.R
import me.amadeucavalcante.a2048.databinding.ActivityMainBinding
import me.amadeucavalcante.a2048.viewmodel.GameViewModel

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val viewModel = GameViewModel()

        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.viewModel = viewModel
    }
}
